official/mnist/mnist_test.py

```diff
@@ -18,52 +18,76 @@
 from __future__ import print_function
 
 import tensorflow as tf
+import time
 
 import mnist
 
-tf.logging.set_verbosity(tf.logging.ERROR)
+BATCH_SIZE = 100
 
 
-class BaseTest(tf.test.TestCase):
+def dummy_input_fn():
+  image = tf.random_uniform([BATCH_SIZE, 784])
+  labels = tf.random_uniform([BATCH_SIZE], maxval=9, dtype=tf.int32)
+  return image, tf.one_hot(labels, 10)
 
-  def input_fn(self):
-    features = tf.random_uniform([55000, 784])
-    labels = tf.random_uniform([55000], maxval=9, dtype=tf.int32)
-    return features, tf.one_hot(labels, 10)
 
-  def mnist_model_fn_helper(self, mode):
```

Comment:
What was the reason for deleting this test?

Reply:
My thinking was that this was testing an implementation detail, not testing the interface expectations.

For the end-user of the `mnist.py` model, what we want from `model_fn` is to create an `Estimator` that can successfully train, predict, and evaluate. And that the predictions produce a scalar class and a vector of probabilities. And that the evaluation produces ...

The added tests to validate those end expectations. So I wasn't clear on what testing the `model_fn` directly is buying.

Thoughts?

Reply:
model_fn is easier to test. 

However, in a second step I think it would be nice to make this part of the code generic (independent of the model_fn), and actually test it with a mocked out model_fn.

Reply:
`model_fn` is insufficient to test no? For example, testing the output of `model_fn` won't result in executing the graph, so we won't catch any errors thrown at runtime because say the graph was invalid.

As discussed offline, it seems we want to keep the testing proposed in this PR?

Reply:
sgtm

Reply:
I agree that adding the new test is good, but I'm not sure there's anything wrong with this one. I believe everything it checks for are things that the model_fn should abide by. Is there an assert in here that you think is bad?

Reply:
Restored the tests, PTAL